Docs for `BackupsIOThreadPool` 2/2 (#47300)
ClickHouse/ClickHouse: docs/en/operations/server-configuration-parameters/settings.md

<thread_pool_queue_size>12000</thread_pool_queue_size>
```

## max_io_thread_pool_size {#max-io-thread-pool-size}

ClickHouse uses threads from the IO Thread pool to do some IO operations (e.g. to interact with S3). `max_io_thread_pool_size` limits the maximum number of threads in the pool.

Possible values:

-   Positive integer.

Default value: `100`.

## max_io_thread_pool_free_size {#max-io-thread-pool-free-size}

> If the number of **idle** threads in the IO Thread pool is greater than `max_io_thread_pool_free_size`, then ClickHouse releases resources occupied by some threads and the pool size is decreased. Threads can be created again if necessary.

Reviewer: ```suggestion
If the number of **idle** threads in the IO Thread pool exceeds `max_io_thread_pool_free_size`, ClickHouse will release resources occupied by idling threads and decrease the pool size. Threads can be created again if necessary.
```

Possible values:

-   Positive integer.

Default value: `0`.

## io_thread_pool_queue_size {#io-thread-pool-queue-size}

The maximum number of jobs that can be scheduled on the IO Thread pool.

Possible values:

-   Positive integer.
-   0 — No limit.

Default value: `10000`.

## max_backups_io_thread_pool_size {#max-backups-io-thread-pool-size}

ClickHouse uses threads from the Backups IO Thread pool to do S3 backup IO operations. `max_backups_io_thread_pool_size` limits the maximum number of threads in the pool.

Possible values:

-   Positive integer.

Default value: `1000`.

## max_backups_io_thread_pool_free_size {#max-backups-io-thread-pool-free-size}

> If the number of **idle** threads in the Backups IO Thread pool is greater than `max_backup_io_thread_pool_free_size`, then ClickHouse releases resources occupied by some threads and the pool size is decreased. Threads can be created again if necessary.

Reviewer: ```suggestion
If the number of **idle** threads in the Backups IO Thread pool exceeds `max_backup_io_thread_pool_free_size`, ClickHouse will release resources occupied by idling threads and decrease the pool size. Threads can be created again if necessary.
```
